src/System.Private.CoreLib/shared/System/Security/PermissionSet.cs

```diff
@@ -18,38 +19,38 @@ public partial class PermissionSet : ICollection, IEnumerable, IDeserializationC
+        public virtual PermissionSet? Copy() { return default(PermissionSet); }
         public virtual void CopyTo(Array array, int index) { }
         public void Demand() { }
         [Obsolete]
         public void Deny() { throw new PlatformNotSupportedException(SR.PlatformNotSupported_CAS); }
-        public override bool Equals(object o) => base.Equals(o);
+        public override bool Equals(object? o) => base.Equals(o);
         public virtual void FromXml(SecurityElement et) { }
         public IEnumerator GetEnumerator() { return GetEnumeratorImpl(); }
         protected virtual IEnumerator GetEnumeratorImpl() { return Array.Empty<object>().GetEnumerator(); }
         public override int GetHashCode() => base.GetHashCode();
-        public IPermission GetPermission(Type permClass) { return GetPermissionImpl(permClass); }
-        protected virtual IPermission GetPermissionImpl(Type permClass) { return default(IPermission); }
-        public PermissionSet Intersect(PermissionSet other) { return default(PermissionSet); }
+        public IPermission? GetPermission(Type permClass) { return GetPermissionImpl(permClass); }
+        protected virtual IPermission? GetPermissionImpl(Type permClass) { return default(IPermission); }
+        public PermissionSet? Intersect(PermissionSet other) { return default(PermissionSet); }
         public bool IsEmpty() { return false; }
         public bool IsSubsetOf(PermissionSet target) { return false; }
         public bool IsUnrestricted() { return false; }
         public void PermitOnly() { throw new PlatformNotSupportedException(SR.PlatformNotSupported_CAS); }
-        public IPermission RemovePermission(Type permClass) { return RemovePermissionImpl(permClass); }
-        protected virtual IPermission RemovePermissionImpl(Type permClass) { return default(IPermission); }
+        public IPermission? RemovePermission(Type permClass) { return RemovePermissionImpl(permClass); }
+        protected virtual IPermission? RemovePermissionImpl(Type permClass) { return default(IPermission); }
         public static void RevertAssert() { }
-        public IPermission SetPermission(IPermission perm) { return SetPermissionImpl(perm); }
-        protected virtual IPermission SetPermissionImpl(IPermission perm) { return default(IPermission); }
+        public IPermission? SetPermission(IPermission perm) { return SetPermissionImpl(perm); }
+        protected virtual IPermission? SetPermissionImpl(IPermission perm) { return default(IPermission); }
         void IDeserializationCallback.OnDeserialization(object sender) { }
-        public override string ToString() => base.ToString();
-        public virtual SecurityElement ToXml() { return default(SecurityElement); }
-        public PermissionSet Union(PermissionSet other) { return default(PermissionSet); }
+        public override string? ToString() => base.ToString();
```

Comment:
should instead the concrete implementation add constrain then?

Reply:
> should instead the concrete implementation add constrain then?

The current plan is no because it's virtual and overrides do return null, so consumers need to be able to handle null when invoked from the base.

Reply:
@stephentoub now I'm confused :smile: So do you prefer this to be `string?` or `string`? (first message suggests `string` the second `string?`)

Reply:
The override should be `string` even though the base is `string?`.